tests/e2e/pipelinerollout_test.go

```diff
@@ -0,0 +1,280 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/json"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	numaflowv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+
+	apiv1 "github.com/numaproj/numaplane/pkg/apis/numaplane/v1alpha1"
+)
+
+var pipelineSpecSourceRPU = int64(5)
+var pipelineSpecSourceDuration = metav1.Duration{
+	Duration: time.Second,
+}
+var pipelineSpec = numaflowv1.PipelineSpec{
+	InterStepBufferServiceName: "my-isbsvc",
+	Vertices: []numaflowv1.AbstractVertex{
+		{
+			Name: "in",
+			Source: &numaflowv1.Source{
+				Generator: &numaflowv1.GeneratorSource{
+					RPU:      &pipelineSpecSourceRPU,
+					Duration: &pipelineSpecSourceDuration,
+				},
+			},
+		},
+		{
+			Name: "out",
+			Sink: &numaflowv1.Sink{
+				AbstractSink: numaflowv1.AbstractSink{
+					Log: &numaflowv1.Log{},
+				},
+			},
+		},
+	},
+	Edges: []numaflowv1.Edge{
+		{
+			From: "in",
+			To:   "out",
+		},
+	},
+}
+
+var _ = Describe("PipelineRollout e2e", func() {
+
+	const (
+		pipelineRolloutName = "e2e-pipeline-rollout"
+	)
+
+	pipelinegvr := getGVRForPipeline()
+
+	It("Should create the PipelineRollout if it does not exist", func() {
+
+		pipelineRolloutSpec := CreatePipelineRolloutSpec(pipelineRolloutName, Namespace)
+		_, err := pipelineRolloutClient.Create(ctx, pipelineRolloutSpec, metav1.CreateOptions{})
+		Expect(err).ShouldNot(HaveOccurred())
+
+		By("Verifying that the PipelineRollout was created")
+		Eventually(func() error {
+			_, err := pipelineRolloutClient.Get(ctx, pipelineRolloutName, metav1.GetOptions{})
+			return err
+		}).WithTimeout(testTimeout).Should(Succeed())
```

Comment:
Why did you decided to use `WithTimeout` and not pass a timeout and interval to the Eventually func as arguments?

Reply:
Personal preference, I just like the readability.